icu4c/source/i18n/regexst.h

```diff
@@ -39,17 +40,16 @@ class RegexStaticSets : public UMemory {
     RegexStaticSets(UErrorCode *status);         
     ~RegexStaticSets();
     static void    initGlobals(UErrorCode *status);
-    static UBool   cleanup();
 
-    UnicodeSet    *fPropSets[URX_LAST_SET];     // The sets for common regex items, e.g. \s
-    Regex8BitSet   fPropSets8[URX_LAST_SET];    // Fast bitmap sets for latin-1 range for above.
+    UnicodeSet    fPropSets[URX_LAST_SET] {};      // The sets for common regex items, e.g. \s
+    Regex8BitSet  fPropSets8[URX_LAST_SET] {};     // Fast bitmap sets for latin-1 range for above.
 
-    UnicodeSet    fRuleSets[10];               // Sets used while parsing regexp patterns.
-    UnicodeSet    fUnescapeCharSet;            // Set of chars handled by unescape when
-                                               //   encountered with a \ in a pattern.
-    UnicodeSet    *fRuleDigitsAlias;
-    UText         *fEmptyText;                 // An empty string, to be used when a matcher
-                                               //   is created with no input.
+    UnicodeSet    fRuleSets[kRuleSet_count] {};    // Sets used while parsing regexp patterns.
```

Comment:
This used to be an array of 10 elements, but is now only 3. Is this OK?

Reply:
Yes, the 3 is computed at build time to be the actual size required.